cpp/src/arrow/acero/swiss_join.cc

```diff
@@ -494,10 +495,13 @@ void RowArrayMerge::CopyFixedLength(RowTableImpl* target, const RowTableImpl& so
     int64_t num_words_per_row = fixed_length / sizeof(uint64_t);
     for (int64_t i = 0; i < num_source_rows; ++i) {
       int64_t source_row_id = source_rows_permutation[i];
+      DCHECK_LE(source_row_id, std::numeric_limits<uint32_t>::max());
```

Comment:
If that's always the case then are we wasting memory and CPU cache by having a 64-bit permutation array?

Reply:
We are. In fact I have another WIP branch trying to clean them up.